### PR TITLE
Fix teleport crash.

### LINKED ISSUE #29484: Segmentation Fault when teleporting
**Describe the bug**  
The game instantly crashes with a segment fault when using the long distance debug teleport. This isn't game breaking and will probably never happen during a normal play session. But not being able to teleport around in Debug Mode is pretty inconvenient.

**To Reproduce**  
1. Teleport somewhere with the Debug Menu

**Screenshots**  
image

**Versions and configuration(please complete the following information):**
 - OS: Win 10
 - Game Version: Expirimental Build 8765
 - Graphics version: Tiles
 - Mods loaded: Defaults

### LINKED ISSUE #29681: Unexpected Termination Segmentation Fault crashes
[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/3095599/crash.log)
[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/3095600/crash.log)

Viewing certain tiles will cause the game to crash. This happens on multiple builds, in multiple locations, in multiple worlds. The worlds will be fine for a few hours and then a certain tile will just crash the game when viewed. 

**To Reproduce**  
1. Start world
2. Go about making a nice base
3. Play for a few hours
4. Now every time you enter the south corner of the second floor of your base (just an example) the game crashes. It doesn't matter if you teleport there, go up or down stairs, or walk there.

**Expected behavior**  
You enter the area, the game crashes and throws up these 2 messages.

[
image
](url)

[
image
](url)

**Versions and configuration(please complete the following information):**
 - OS: Windows 10 professional
 - Game Version: 0.D-1921-gbe0d520
 - Graphics version: Tiles
 - Mods loaded: Extended realistic guns, Makeshift items, More survival tools, Modular turrets, Salvaged robots, Crazy cataclysm, Beta camp, More locations, Fuji's buildings, More city locations, urban development, Boats, folding parts, vehicle additions, tanks, magic items, necromancy, Jury-rigged robots, classes and scenarios.

I doubt it's any of these mods as I have the problem without them.

**Additional context**  
Save file (To get the crash go down the stairs nearby)
[Big Springs.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3095239/Big.Springs.zip)

UPDATE: They're getting worse. Don't even need to enter a new area anymore. Just going to sleep is causing them. 

[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/3095601/crash.log)

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
``SUMMARY: None```

#### Purpose of change
Fixes #29681 
Fixes #29484 
Fixes #29409 
Basically it's a crash fix, some of these are triggered by the exact scenario I used to reproduce this, and some simply started happening at about the same time which makes me think they have the same cause.

#### Describe the solution
Making these lambda static does something I don't believe the author intended, which is that the first time they are executed, they capture various pieces of their environment, and never do so again.  Once the game data is sufficiently altered, the captured data is invalidated, and accessing it starts risking a crash.
In other words, don't stash lambdas in static variables if they have a capture list, especially [&].

### CODE REVIEW

**Comment by scrubs2009** [comment]
[#29681](https://github.com/CleverRaven/Cataclysm-DDA/issues/29681) has not been fixed. Same error still occurs.
